### PR TITLE
Qute: make it possible to supply a template backed by a build item

### LINKED ISSUE #41386: Qute: make it possible to supply a template backed by a build item
### Description

Currently, an extension can only supply a template from a file located in `src/main/resources/templates`.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #41386

### CODE REVIEW

**Comment by ia3andy** [comment]
That was fast :)

**Comment by ia3andy** [comment]
I will have a look tomorrow :)

**Review by gastaldi — APPROVED** [review]

**Comment by mkouba** [comment]
> I will have a look tomorrow :)

@ia3andy did you get a chance to look at it? ;-)

File: extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplatePathBuildItem.java [hunk 1]
```diff
@@ -1,28 +1,46 @@
...
 
     static final String TAGS = "tags/";
 
     private final String path;
-    private final Path fullPath;
     private final String content;
+    private final Path fullPath;
+    private final String extensionInfo;
+
+    public TemplatePathBuildItem(String path, String extensionInfo, String content) {
```
Comment (ia3andy) [inline extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplatePathBuildItem.java:25]: @mkouba could you create a builder and add a bit of doc for the different fields?

Comment (mkouba) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplatePathBuildItem.java:25]: Sure, why not ;-)

Comment (mkouba) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplatePathBuildItem.java]: Done.

**Review by ia3andy — APPROVED** [review]
Thanks this is awesome, I tested it in https://github.com/quarkiverse/quarkus-roq/pull/24

**Pull request merged**